LDEventSource/LDEventParser.m

```diff
@@ -66,14 +79,57 @@ -(LDEvent*)eventFromString:(NSString*)eventString {
                 } else if ([key isEqualToString:LDEventIDKey]) {
                     event.id = value;
                 } else if ([key isEqualToString:LDEventRetryKey]) {
-                    self.foundRetryInterval = [value doubleValue];
+                    if ([value isKindOfClass:[NSNumber class]]) {
+                        self.retryInterval = @([value doubleValue]);
+                    }
                 }
             }
         }
     }
-    if (self.remainingEventString.length == 0) {
-        self.remainingEventString = nil;
-    }
-    return event;
+}
+
+//extracts lines from the first thru the first empty line
+-(NSArray<NSString*>*)linesToParseFromEventString {
+    if (self.eventString.length == 0) { return nil; }
+
+    NSArray<NSString*> *lines = [self eventStringLines];
+    NSUInteger indexOfFirstEmptyLine = [self indexOfFirstEmptyEventStringLine];
+    if (indexOfFirstEmptyLine == NSNotFound) { return nil; }
+
+    NSArray<NSString*> *linesToParse = [lines subarrayWithRange:NSMakeRange(0, indexOfFirstEmptyLine + 1)];
+    if (linesToParse.count == 0) { return nil; }
+
+    return linesToParse;
+}
+
+-(NSString*)remainingEventStringAfterParsingEventString {
+    if (self.eventString.length == 0) { return nil; }
+
+    NSArray<NSString*> *lines = [self.eventString componentsSeparatedByCharactersInSet:[NSCharacterSet newlineCharacterSet]];
```

Comment:
#picky we do this enough times, I wonder if we should add it as a category method to NSString?